### PR TITLE
Tests

### PR DESCRIPTION
**Pull request by mruwnik**
I accidentally merged the previous tests PR to the wrong branch, so this has an extra commit containing the coda tests, along with the parser tests

### CODE REVIEW

File: parser/__tests__/coda.test.js [hunk 1]
```diff
@@ -0,0 +1,195 @@
+import { jest } from "@jest/globals";
```
Comment (mruwnik) [inline parser/__tests__/coda.test.js:1]: these are the same as in the previous PR

File: parser/__tests__/coda.test.js [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+import fetch from "jest-fetch-mock";
+import { getAnswers, getDocIDFromLink, updateAnswer } from "../coda.js";
+import { codaColumnIDs } from "../constants.js";
+
+fetch.enableMocks();
+
+describe("getDocIDFromLink", () => {
+  it("returns the correct doc ID for a valid Google Docs link", () => {
```
Comment (mruwnik) [inline parser/__tests__/coda.test.js:9]: other than all `test(...)` being changed to `it(...)` :D

File: parser/parser.js [hunk 3]
```diff
@@ -1,14 +1,13 @@
-// Accumulators for the count and total text length of all suggestions
-let suggestions = new Map();
-
-export const parseDoc = (doc) => {
+// TODO: test this function. Can't be bothered to now...
```
Comment (mruwnik) [inline parser/parser.js:1]: this...

File: parser/parser.js [hunk 4]
```diff
@@ -1,14 +1,13 @@
...
+export const parseDoc = async (doc) => {
   // contextual information about the doc that is sometimes useful
   // to the parsers of particular elements
   const documentContext = {
     footnotes: doc.footnotes || {},
     namedStyles: doc.namedStyles,
     inlineObjects: doc.inlineObjects,
     lists: doc.lists || {},
+    suggestions: new Map(), // Accumulators for the count and total text length of all suggestions
```
Comment (mruwnik) [inline parser/parser.js:10]: this was being cleared with each call of this function, which
a) would be interesting with async functions
b) makes testing it really hard
c) is ugly
so I moved it into this object

File: parser/parser.js [hunk 5]
```diff
@@ -159,40 +157,11 @@ const getLWTag = (tagName) => {
...
-  return md
-    .replace(footnotePattern, footnoteReplacement)
-    .replace(footnoteRefPattern, footnoteRefReplacement);
-};
+const getLWTag = (tagName) => getTag("https://www.lesswrong.com", tagName);
+const getEAFTag = (tagName) =>
+  getTag("https://forum.effectivealtruism.org", tagName);
 
-const parseParagraph = (documentContext) => (paragraph) => {
+export const parseParagraph = (documentContext) => (paragraph) => {
```
Comment (mruwnik) [inline parser/parser.js:164]: I tried various ways to get around private helper functions not being available for tests, but gave up and just exported them. It's annoying that there's no way to have them generally private, but testable. There's no way I'm going to mess about with parameters to the main `parseDoc()` function just to test how each of these smaller ones works...

File: .github/workflows/tests.yaml [hunk 6]
```diff
@@ -0,0 +1,23 @@
+name: Run tests
```
Comment (mruwnik) [inline .github/workflows/tests.yaml:1]: this already gets triggered - nice!

File: parser/parser.js (hunk 5 above)
Comment (Aprillion) [reply parser/parser.js:164]: if you are afraid someone would start using these wanna-be-private-but-exported-for-tests functions by accident, you can wrap stuff into a scary-named object:
```js
const parseParagraph = ...
...
export const _ = {parseParagraph, parsetextRun, ...}
```

File: .github/workflows/tests.yaml [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+
+on: [push, pull_request]
```
Comment (Aprillion) [inline .github/workflows/tests.yaml:3]: can we choose 1 trigger without repeating the same test twice?

```suggestion
on: pull_request
```

**Review by Aprillion — APPROVED** [review]

**Pull request merged**